### ISSUE TITLE
cloneSheet function

### ISSUE THREAD

**Issue by jkeane82** [miraisolutions/xlconnect#68]
Using R (R 3.3), XLConnect package V 0.2.12, rJava package V 0.9.8, receiving error.

Error: IllegalArgumentException (Java): Sheet index (-1) is out of range (0..3)

when I run clonesheet function. Below is the code for the workbook:

wb <- loadWorkbook(paste0("C:\\Users\\james.keane\\Documents\\R\\RETC R 
    Scripts\\Client Progress Report\\", fileName, separate = ""), create = 
    TRUE)
setStyleAction(wb,XLC$"STYLE_ACTION.NONE")        # This allows the 
     spreadsheet to be populated with data without chaninging the formating 
     of the spreadsheet

tabNames <- getSheets(wb)
if(length(grep(taxSeason, tabNames)) >= 1){
     writeWorksheet(wb, paste(taxSeason, "Property Tax Appeal Status", 
         separate = " "), sheet = taxSeason, startRow = 1, startCol = 1, 
         header = FALSE)
     writeWorksheet(wb, reportClient[,1:22], sheet = taxSeason, startRow = 
         3, startCol = 1, header = FALSE)
     writeWorksheet(wb, reportClient[,28:33], sheet = taxSeason, startRow = 
          3, startCol = 28, header = FALSE)
}else{
     cloneSheet(wb, "Template", taxSeason)
     writeWorksheet(wb, paste(taxSeason, "Property Tax Appeal Status", 
         separate = " "), sheet = taxSeason, startRow = 1, startCol = 1, 
         header = FALSE)
     writeWorksheet(wb, reportClient[,1:22], sheet = taxSeason, startRow = 
         3, startCol = 1, header = FALSE)
     writeWorksheet(wb, reportClient[,28:33], sheet = taxSeason, startRow = 
         3, startCol = 28, header = FALSE)
}
saveWorkbook(wb)

Once the workbook is saved and the file is opened I'll have the following tabs appear: "Exhibit A", "Template", "Template(2)", "Data Validation". Before the code is run, there are only the three tabs "Exhibit A", "Template", and "Data Validation". Once the code is run, it should clone the Template tab, rename it to the appropriate taxSeason year, and then this new sheet is populated with data using the writeWorkSheet function.

There have been times that it has worked and others it doesn't so not sure why it's going back and forth. Any ideas on why this is occurring?

**Comment by martinstuder** [comment]
Hi,

do you have a minimal example (code & file) that reproduces the issue? The exception that you get basically says that it can't find a worksheet with a certain name. To understand in detail what is going on I would need a sample file though.

Best regards,
Martin

**Comment by jkeane82** [comment]
Hi Martin,

Thank you for taking the time to respond.  I believe the issue was that the file I was calling was corrupt.  I created a new file with the same tabs mentioned in my description and it now works perfectly.  But I will say that it was odd that when having the error, once I saved the workbook and opened the file, there would be a new tab but with the same name and a number in parentheses.  So, it was finding the tab to clone, but for some reason it couldn't assign it the new name due to the file being corrupted.  At least that's what I believe at the moment.  If I have any issues I'll come back to this forum and give a simple example for you to test it.

Again, thank you for taking the time to respond.

Take Care,
James
